pouta_blueprints/drivers/provisioning/openshift_driver.py

```diff
@@ -0,0 +1,638 @@
+import base64
+import json
+import time
+import uuid
+from pprint import pprint
+import requests
+# noinspection PyUnresolvedReferences
+from six.moves.urllib.parse import urlparse, parse_qs
+
+from pouta_blueprints.client import PBClient
+from pouta_blueprints.drivers.provisioning import base_driver
+
+
+class OpenShiftClient(object):
+    def __init__(self, base_url, subdomain, user, password):
+        if base_url[-1] == '/':
+            base_url = base_url[:-1]
+        self.base_url = base_url
+        self.subdomain = subdomain
+        self.oapi_base_url = base_url + '/oapi/v1'
+        self.kube_base_url = base_url + '/api/v1'
+        self.user = user
+        self.password = password
+
+        self.token_data = None
+        self._session = requests.session()
+
+    @staticmethod
+    def make_base_kube_object(kind, name=None):
```

Comment:
Yes, inherited from Kubernetes. It is the object type, always present in object descriptions.